File: programmers/lv1/176963.py
# https://school.programmers.co.kr/learn/courses/30/lessons/176963?language=python
def solution(name, yearning, photo):
    d = {name: score for name, score in zip(name, yearning)}
    return [sum(d.get(person, 0) for person in people) for people in photo]

File: programmers/lv1/176963.scala
/* https://school.programmers.co.kr/learn/courses/30/lessons/176963?language=scala */
object Solution {
    def solution(
        name: Vector[String],
        yearning: Vector[Int],
        photo: Vector[Vector[String]]
    ): Vector[Int] = {
        val scoreMap: Map[String, Int] = name.zip(yearning).toMap

        photo.map { people =>
            people.map(person => scoreMap.getOrElse(person, 0)).sum
        }
        # return Vector[Int]()
    }
}